tests/Integration/Classes/ObjectModelTest.php

```diff
@@ -107,11 +107,11 @@ public function testTrans(): void
         $newObject = new TestableObjectModel();
         $transMethod = new \ReflectionMethod($newObject, 'trans');
         $transMethod->setAccessible(true);
-        $trans = $transMethod->invoke($newObject, '<a href="test">%d Succesful deletion "%s"</a>', ['_raw' => true, 10, '<b>stringTest</b>'], 'Admin.Notifications.Success');
+        $trans = $transMethod->invoke($newObject, '<a href="test">%d Succesful deletion "%s"</a>', [10, '<b>stringTest</b>'], 'Admin.Notifications.Success');
         $this->assertEquals('<a href="test">10 Succesful deletion "<b>stringTest</b>"</a>', $trans);
 
-        $trans = $transMethod->invoke($newObject, '<a href="test">%d Succesful deletion "%s"</a>', [10, '<b>stringTest</b>'], 'Admin.Notifications.Success');
-        $this->assertEquals('&lt;a href="test"&gt;10 Succesful deletion "&lt;b&gt;stringTest&lt;/b&gt;"&lt;/a&gt;', $trans);
+        $trans = $transMethod->invoke($newObject, '<a href="test">%d Succesful deletion "%s"</a>', [10, htmlspecialchars('<b>stringTest</b>')], 'Admin.Notifications.Success');
+        $this->assertEquals('<a href="test">10 Succesful deletion "&lt;b&gt;stringTest&lt;/b&gt;"</a>', $trans);
```

Comment:
This test should normally look like this for XSS to be mitigated, so that when you want to output html tags, you can do so in parameters
```php
        $trans = $transMethod->invoke($newObject, '<a href="test">%d Succesful deletion "%s"</a>', [10, '<b>stringTest</b>'], 'Admin.Notifications.Success');
        $this->assertEquals('&lt;a href="test"&gt;10 Succesful deletion "<b>stringTest</b>"</a>', $trans);
```

Reply:
@Tofandel This PR is the final result of years long discussion 😄 see https://github.com/PrestaShop/PrestaShop/pull/30415#pullrequestreview-1196406957

Yes it introduces a security issue but this is because it fixes a design issue. A necessary security step was added in the wrong place and we now remove it from this wrong place. The escaping should not happen inside `trans()`.

Reply:
> The escaping should not happen inside trans()

It should, but the escaping is indeed in the wrong place, you cannot be sure that what you are getting from a translation is safe

So to achieve that you escape the output of the retrieved translation but do not escape the parameters

This is the only way to achieve XSS safety

Reply:
`trans()` is a function that is supposed to translate strings. It has nothing to do with security. Security should not be handled inside `trans()`.

Escaping properly HTML before sending it to user is not the responsibility of the translator, it is the responsibility of the controller/view layer.

> I can provide a super simple XSS demo with this, and it should not be released like this, that's a 7 out of 10 CVE because it allows translators of a site to inject html anywhere they want and then gain unauthorized access by data exfiltration

From the Back-office, it is possible to upload a SVG image which can carry a XSS too. Translators can only "inject" HTML if they provide translations that are being imported into PrestaShop from the Back-office. That means they have been trusted by the Back-office administrator to be granted access to the Back-office or he trusted them when they provided translation files. If they have been trusted but they use this trust to inject HTML, the problem is that the Back-office administrator trusted the wrong persons, it's not the Back-office.

You can see translation files as inputs just like modules. If a merchant installs a garbage module, it could crash his shop, it could expose vulnerable code that makes his shop vulnerable. The problem is not that PrestaShop did not "prevent" or "validate" the wrong module: installing only trusted modules is the responsibility of the shop owner.

Reply:
I'll have to disagree, in big companies access control is very important. It's even common to outsource translation to freelancers that should only be given permission to the translation in the back office

But if now they start to also have the ability to inject scripts transparently wherever they please without anyone noticing, effectively installing backdoors then there is a big problem

Translation should never allow html in from the translator side, so it should be escaped. The output of the translation though shouldn't be escaped

If html needs to be used for links or other, that's when the placeholders and parameters come in handy

Reply:
This PR is the last step of a discussion and analyzis that has been going on since October 2022 so please trust us when I tell you it's the best compromise we could get 😉 

Escaping HTML? It's a good thing.
Doing it inside `trans()` ? It's not a good idea. It brings you hundred of other problems.

The idea of this PR is not "let's remove HTML escaping" it's "let's move the HTML escaping to the right place".

If you do not understand the reasons behind it, I can show you the history:
- bug https://github.com/PrestaShop/PrestaShop/issues/29940
- 1st attempt to fix it, this is where idea to remove escape from `trans()`  https://github.com/PrestaShop/PrestaShop/pull/29943
- related problem https://github.com/PrestaShop/PrestaShop/pull/29584
- PR https://github.com/PrestaShop/PrestaShop/pull/30415 for 8.0.0
- Issue https://github.com/PrestaShop/PrestaShop/issues/30539

Sorry I don't want to continue writing hundred words of explanation ^^ this is a topic that span for 8 months now so I have not enough motivation to tell you every aspects of it that brought us to this conclusion.

If, after reading all of the history, you can provide an interesting solution that meets all criteria, including fixing bug without breaking something else, then I'll be happy to review it 😉

Reply:
Hi @Tofandel I just wanted to tell you I don't forget you, this message is inside my todolist and I have requested the help of someone else on this topic because I can't handle everything.